### PR TITLE
ICU-22583 BRS74: Version number update for 74.2

### PR DESCRIPTION
**Pull request by mihnita**
##### Checklist

- [x] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-22583
- [x] Required: The PR title must be prefixed with a JIRA Issue number. 
- [x] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [x] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [x] Issue accepted (done by Technical Committee after discussion)
- [x] Tests included, if applicable
- [x] API docs and/or User Guide docs changed or added, if applicable

ALLOW_MULTIPLE_COMMITS=true

### CODE REVIEW

**Comment by mihnita** [comment]
Differences between this and PR#2728:
* this also updates `icu/dev/test/util/DebugUtilitiesData.java`
* this also updates `tools/cldr/cldr-to-icu/build-icu-data.xml`
* `icu4j/releases_tools/shared.sh` uses `74.2` for `github_rel_version` (instead of `74_2`)
    This is to match the naming done "by hand" in relase 74.1 for all the Java files (.jars)
    See for example https://github.com/unicode-org/icu/releases/tag/release-74-1.
    The initial script used `_` to be consistent with the C/C++ files and release [73.1](https://github.com/unicode-org/icu/releases/tag/release-73-1).
     But Yoshito used `.` for consistency with [Maven Central](https://repo1.maven.org/maven2/com/ibm/icu/icu4j/74.1/)

File: icu4c/source/data/misc/icuver.txt
Comment (markusicu) [inline icu4c/source/data/misc/icuver.txt:1]: What about rebuilding https://github.com/unicode-org/icu/blob/maint/maint-74/icu4j/main/core/src/main/resources/com/ibm/icu/impl/data/icudt74b/icuver.res ?

**Comment by mihnita** [comment]
I don't know how to solve
   `"continuous-integration/appveyor/pr — AppVeyor was unable to build non-mergeable pull request"`
What is that?

**Comment by markusicu** [comment]
> I don't know how to solve `"continuous-integration/appveyor/pr — AppVeyor was unable to build non-mergeable pull request"` What is that?

It's one of two CI systems (besides Travis) that we used years ago but which became slow and flaky. [We moved those CI tasks to Azure CI](https://github.com/search?q=repo%3Aunicode-org%2Ficu+appveyor&type=commits) and maybe also to GHA in 2021.

I don't know why it's popping up now. Some kind of zombie... it's not a required check, so I think we can ignore it.

However, please look at my earlier question.

**Comment by mihnita** [comment]
> What about rebuilding https://github.com/unicode-org/icu/blob/maint/maint-74/icu4j/main/core/src/main/resources/com/ibm/icu/impl/data/icudt74b/icuver.res ?


Done, but things are ugly :-(

There are 204 files changed:
* icu4j/main/core/src/main/resources/com/ibm/icu/impl/data/icudt74b/icuver.res 
* icu4j/main/core/src/main/resources/com/ibm/icu/impl/data/icudt74b/unit/res_index.res 
* icu4j/main/core/src/main/resources/com/ibm/icu/impl/data/icudt74b/zone/res_index.res 
* icu4j/main/currdata/src/main/resources/com/ibm/icu/impl/data/icudt74b/curr/res_index.res 
* icu4j/main/langdata/src/main/resources/com/ibm/icu/impl/data/icudt74b/lang/res_index.res 
* icu4j/main/regiondata/src/main/resources/com/ibm/icu/impl/data/icudt74b/region/res_index.res 
* 190 x main/charset/src/main/resources/com/ibm/icu/impl/data/icudt74b/*.cnv
* 8 x main/core/src/test/resources/com/ibm/icu/dev/data/testdata/*.cnv

I hex-dumped and diff-ed, to understand what's going on.

The `res_index` files changed because the list of locales changed 2 days ago, adding `kxv_Latn` :
https://github.com/unicode-org/icu/pull/2727/commits
(the last commit, part 4, which restored the kxv_Latn*.txt files)

The .cnv files contain the ICU version.
```
00000000  00 20 da 27 00 14 00 00  01 00 02 00 63 6e 76 74  |. .'........cnvt|
00000010  06 02 00 00 4a 02 00 00  00 00 00 00 00 00 00 00  |....J...........|
                      ^^ ^^
```

**Review by markusicu — APPROVED** [review]
Thanks!
Too bad that there are so many files that carry the version number...
I added the magic incantation that should allow you to merge without squashing.

**Comment by mihnita** [comment]
Thanks.
I'll submit once all tests pass.

**Comment by mihnita** [comment]
Ouch, my bad... I squashed...
Now I have to wait again for all tests to pass :-(

**Comment by markusicu** [comment]
Thank you @mihnita !

**Pull request merged**